include/fsm/fill_spill_merge.hpp

```diff
@@ -539,18 +546,19 @@ static dh_label_t OverflowInto(
   //SECOND PLACE TO STASH WATER: IN THIS DEPRESSION'S NEIGHBOUR
   //Maybe we can fit it into this depression's overflow depression!
 
+
   auto &pdep = deps.at(this_dep.parent);
   if(this_dep.odep==NO_VALUE){      //Does the depression even have such a neighbour? 
-    if(this_dep.parent!=OCEAN && pdep.water_vol==0) //At this point we're full and heading to our parent, so it needs to know that it contains our water
-      pdep.water_vol += this_dep.water_vol;
+    if(this_dep.parent!=OCEAN && pdep.water_vol==0 && (pdep.lchild == this_dep.dep_label || pdep.rchild == this_dep.dep_label)) //At this point we're full and heading to our parent, so it needs to know that it contains our water
```

Comment:
Why?

Reply:
I'm not 100% certain if this one is strictly necessary, but it made sense to me. What if we reach this point with a depression that is ocean-linked to its parent? If this is the case, the parent shouldn't receive the water volume.

Reply:
But the parent should receive that water, because the oceanlink will be passing water to the place it needs to go.

Reply:
Hang on - surely all depressions have an odep? Then we would be nitpicking over an if statement that never executes.